### PR TITLE
[Clown Traitor Item] Bikehorn esword

### PR DESCRIPTION
**Pull request by Time-Green**
A clown can now buy an esword disguised as a bikehorn through his uplink for 7 tc. (Same price as normal eswords)

It actives through alt-clicking.

It's 9 tc instead of 7

:cl:
rscadd: adds a new clown traitor item: A bikehorn esword. Activate it by alt-clicking.
/:cl:

ehonk

### CODE REVIEW

**Comment by Ratvar** [comment]
Foresee security, assistants, scientists and mimes harmbatoning clown for "illegal harrasment" and accidently clicking his bikehorn with alt button on every minute

**Comment by Amelia0010** [comment]
image
that, as well as making it activate on use, like a normal e-sword

**Comment by Partheo** [comment]
Yes,a honk sound on activation is crucial. And pink esword? Yes.

**Comment by Ratvar** [comment]
Make people without clumsyness honk their hand off if they try to activate it.

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
Traitor clowns are free of clumsiness. 

You should mix the esword activation sound and honking somehow

**Comment by Time-Green** [comment]
I'm fine with making it pink, but I'm not letting it activate like a normal esword or have it fucking honk on slashing someone

**Comment by ghost** [comment]
what the fuck, time green

why

**Comment by Time-Green** [comment]
Do people really want it to honk on hit?

**Comment by Partheo** [comment]
yes

**Comment by ghost** [comment]
YES ITS THE FUCKING CLOWN

**Comment by ShadowDeath6** [comment]
make it honk on attack as well as the normal esword sound if you don't want to make it cost more, since making it sound like a normal horn would be a huge buff

**Comment by Time-Green** [comment]
Right, updated.
Now it's pink, honks on slashing and instead of alt activating it, it toggles wheter activating extends the blade or honks
Also it's 9 tc

File: code/game/objects/items/weapons/melee/energy.dm [hunk 1]
```diff
@@ -71,6 +71,9 @@
 	return 0
 
 /obj/item/weapon/melee/energy/attack_self(mob/living/carbon/user)
+	activate(user)
+
+/obj/item/weapon/melee/energy/proc/activate(mob/living/carbon/user)
 	if(clumsy_check && (user.disabilities & CLUMSY) && prob(50))
 		user << "<span class='warning'>You accidentally cut yourself with [src], like a doofus!</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/melee/energy.dm]: to_chat

File: code/game/objects/items/weapons/melee/energy.dm [hunk 2]
```diff
@@ -96,13 +96,24 @@
 		throw_speed = initial(throw_speed)
 		if(attack_verb_on.len)
 			attack_verb = list()
-		icon_state = initial(icon_state)
+		setIcon()
 		w_class = initial(w_class)
 		playsound(user, 'sound/weapons/saberoff.ogg', 35, 1)  //changed it from 50% volume to 35% because deafness
 		user << "<span class='notice'>[src] can now be concealed.</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/melee/energy.dm]: to_chat

File: code/game/objects/items/weapons/melee/energy.dm [hunk 3]
```diff
@@ -226,3 +237,66 @@
...
+
+
+
+/obj/item/weapon/melee/energy/sword/bikehorn/AltClick(mob/living/carbon/L)
+	if(L.mind.special_role)
+		canSword = !canSword
+		if(canSword)
+			L << "<span class='notice'>You squeeze the [name]. Honking it will now extend an esword.</span>"
+		else
+			L << "<span class='notice'>You squeeze the [name]. It functions like a normal bikehorn again.</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/melee/energy.dm]: to_chat

File: code/game/objects/items/weapons/melee/energy.dm [hunk 4]
```diff
@@ -226,3 +237,66 @@
...
+		icon = horn.icon
+		icon_state = horn.icon_state
+
+
+
+/obj/item/weapon/melee/energy/sword/bikehorn/AltClick(mob/living/carbon/L)
+	if(L.mind.special_role)
+		canSword = !canSword
+		if(canSword)
+			L << "<span class='notice'>You squeeze the [name]. Honking it will now extend an esword.</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/melee/energy.dm]: to_chat

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by CrazyDrakkon** [comment]
*sad horn or airhorn on killing blow*

**Comment by Time-Green** [comment]
@AshCorr Merge conflicts were fixed

**Pull request merged**